main.go

```diff
@@ -163,3 +166,10 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func setupWebhooks(mgr ctrl.Manager) {
+	if err := (&infrastructurev1.NestedCluster{}).SetupWebhookWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create webhook", "webhook", "NestedCluster")
+		os.Exit(1)
+	}
+}
```

Comment:
Can we have this if block defined above in `main()` where `setupWebhooks(mgr)` is, the go tag below `// +kubebuilder:scaffold:builder` should actually auto inject this for you if you use `kubebuilder new ...` so to keep the configuration the same place always then when anyone makes changes they always get the same exp.

Reply:
not sure I fully understand the request, anyway, I updated per my understanding

Reply:
Just to reduce turn around cycles on this, all I'm thinking if instead of calling/adding a new func for this put the `if` block back in the `main()` func where you are calling the func. 

The reasoning is the go comment tag is what kubebuilder uses to inject code and one of the piece of code it usually injects is this same exact `if` block.

Reply:
ok, I think I understand the right thing, I updated with additional test, please help check again, thanks !~
@christopherhein